doc: note that @-pattern is accessible in default values (#11672)
NixOS/nix: doc/manual/src/language/syntax.md

    > [ 23 {} ]
    > ```

>     The complete function argument denoted as an `@`-pattern can be accessed inside default values.

Reviewer: "denotes" has a specific meaning (semantics) that is not what we want here. We're just talking about syntax and binding. So let's use words/phrease like "brought in scope" "bound over" etc.

Here's a shot at that:
```suggestion
    The variable bound by the `@`-pattern is also brought into scope over the set pattern on the other side of the `@`, not just over the body of the function.
    It can therefore be used default values:
```

Author: ```suggestion
    The full function argument bound by the `@`-pattern is also brought into scope over the set pattern on the other side of the `@`, not just over the body of the function.
    It can therefore be used in default values:
```

    > [ 23 {} ]
    > ```

>     The full function argument bound by the `@`-pattern is also brought into scope over the set pattern on the other side of the `@`, not just over the body of the function.

Author: ```suggestion
    The full function argument bound by the `@`-pattern is also brought into scope over the set pattern on the other side of the `@`.
```

Maybe the additional contrast is not even necessary, it makes for a very long sentence.

Reviewer 2: Full function argument isn't accurate. It could be read as (original + defaults).
s/full/original would already be better, but how about this?

```suggestion
    All bindings introduced by the function are available in the `?`-defaults, including the original argument from an `@`-pattern:
```

Reviewer 2: ```suggestion
    All bindings introduced by the function are in scope in the entire function expression; not just in the body. For example, the binding of an `@` pattern is available in the `?`-defaults:
```

Reviewer: Ah we can in fact break this up: we can document the general rule above with something like `{ a ? b, b ? a }: ...`, and then document that the same thing also applies to `@` bindings here.

Reviewer 2: The simple case with just `?` is already documented above

Reviewer 2: wait never mind, it's not

Reviewer 3: I found @roberth sentence easier to understand.

>     It can therefore be used in default values:

Author: ```suggestion
    It can therefore be used in default values.
```

>     All bindings introduced by the function are in scope in the entire function expression; not just in the body. For example, the binding of an `@` pattern is available in the `?`-defaults:
>     It can therefore be used in default values.
> 
>     > **Example**
>     >

Author: ```suggestion
    All bindings introduced by the function are in scope in the entire function expression; not just in the body.
    It can therefore be used in default values.

    > **Example**
    >
    >  The binding of an `@` pattern is available in the `?`-defaults:
    >
```
